design/user_controller_workspaces.md

```diff
@@ -28,37 +28,58 @@ I believe this new responsibility could be mostly abstracted away from the user.
 
 ## Pseudo API:
 
-```scala
-master : = lakefs.repo("my-repo").branch("master")
-ws = new Workspace(master)
+```python
+import spark, lakefs
 
-// read
+branch = lakefs.repo("my-repo").branch("main")
+ws = lakefs.workspace(branch, name="daily-ingest-13042021")
+# read
 df = spark.read.parquet(ws.path("my/collection"))
-
-// write
+# write
 df.write.parquet(ws.path("another/location"))
 
-// commit!
+# commit it!
 ws.all().commit("commit everything!")
+# or revert it
+ws.revert()
+```
 
-// or add stuff
-ws.add("**/*.parquet", "**/*.json", "another/location/_SUCCESS")
-    .commit("a commit of only some of the stuff!")
+## Pseudo CLI
 
-// or revert it
-ws.revert()
+```sh
+$ lakectl fs upload ./foo.parquet lakefs://my-repo/main
+No workspace ID found, created one: ozkatz-main-2021031522000
+Upload complete. You can see your uncommitted changes by doing `lakectl diff lakefs://my-repo/main+ozkatz-main-2021031522000` 
+$
+$ lakectl fs upload ./another.parquet lakefs://my-repo/main
+Using workspace ID from ~./lakectl/workspaces/main: ozkatz-main-2021031522000
+Upload complete. You can see your uncommitted changes by doing `lakectl diff lakefs://my-repo/main+ozkatz-main-2021031522000`
+$ lakectl fs commit lakefs://my-repo/main
+Using workspace ID from ~./lakectl/workspaces/main: ozkatz-main-2021031522000
+Commit successful, wrote 2 files, deleted 0. Removing workspace ID: ~./lakectl/workspaces/main
+$ # alternatively, pass an explicit workspace ID
+$ lakectl fs upload ./another.parquet lakefs://my-repo/main+ozkatz-main-2021031522000
+$ # see workspaces per branch
+$ lakectl branch list-workspaces lakefs://my-repo/main
```

Comment:
Where does lakefs know what workspaces there are?  Another file...?  That would require CAS.  Perhaps a "directory", and then we can pretend atomicity matters less.